gui/default/syncthing/core/syncthingController.js

```diff
@@ -588,6 +618,18 @@ angular.module('syncthing.core')
             }).error($scope.emitHTTPError);
         }
 
+        $scope.probeAddress = function (address) {
+            if (address in $scope.cacheGUIAddress) {
+                return $scope.cacheGUIAddress[address]
+            }
```

Comment:
This cache is never cleaned up, so if the ui is running for a long time a lot of stuff might accumulate here.

Perhaps it's possible to capture the content of the current cache in refreshConnectionStats,
any addresses that are still there (when it goes through data from refreshConnectionStats), it's kept there, others are remove (effectively, build a new cache, not hitting things that are in the old cache, and then replace the cache variable with the newly built cache).